[quantization] Re-work Conv2d

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24117 [quantization] JIT serialization for Conv2d
* #24116 [quantization] state_dict serialization for Conv2d + some bugfixes
* **#24115 [quantization] Re-work Conv2d**

Differential Revision: [D16765474](https://our.internmc.facebook.com/intern/diff/D16765474)

Commit:
[quantization] Re-work Conv2d

zafartahirov commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -49,80 +48,48 @@ class Conv2d(_ConvNd):
-    @property
-    def zero_point(self):
-        return self._zero_point.item()
+    def set_weight(self, w):
```
Is there an advantage of using explicit setters/getters instead of the properties? I feel like we are stepping off the common pattern, where weight/bias are just variables, not methods.

dzhulgakov replied:
are properties breaking with TorchScript? but I agree - would be nice to make them work (I think it should be possible to disambiguate properties and fields because properties are on the class level, not object level)

jamesr66a replied:
Yes, properties are unsupported in TorchScript, I can file an issue

Commit:
Update on "[quantization] Re-work Conv2d"


Differential Revision: [D16765474](https://our.internmc.facebook.com/intern/diff/D16765474)

Commit:
Update on "[quantization] Re-work Conv2d"


Differential Revision: [D16765474](https://our.internmc.facebook.com/intern/diff/D16765474)
